### ISSUE TITLE
[FEATURE] Allow provenance generated by CI to be published with a package.

### ISSUE THREAD

**Issue by ianlewis** [npm/cli#6455]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

Currently npm only allows publishing SLSA provenance that was generated by the npm CLI itself.

```shell
npm publish --provenance
```

It would be useful for CI systems to be able to generate provenance themselves and use the `npm` cli to publish the provenance.

### Expected Behavior

The npm cli `--provenance` flag could take an argument that is a path to a [sigstore bundle](https://github.com/sigstore/protobuf-specs/blob/v0.1.0/protos/sigstore_bundle.protob) file in JSON format.

```shell
npm publish --provenance=/path/to/bundle.sigstore
```

This bundle could then be added as provenance metadata when publishing instead of generating provenance itself.

### Steps To Reproduce

N/A

### Environment

N/A

**Comment by ianlewis** [comment]
Support in the CLI is currently blocked by issues with flag parsing where the flag's value could be a boolean or a string.
- #6313

**Comment by ljharb** [comment]
That would be great, as it would mean you don't need to use a blessed CI system to add publish provenance info.

**Comment by ianlewis** [comment]
> That would be great, as it would mean you don't need to use a blessed CI system to add publish provenance info.

Yes, and no. For now you'll still be limited to what can be signed via an OIDC provider. GitHub and perhaps soon Gitlab. But I think that we would like registries to be somewhat agnostic to the builders.

**Comment by ljharb** [comment]
Ah, gotcha. (altho "builders" are irrelevant here since provenance only verifies the publish location/workflow, and nothing more)

**Comment by ianlewis** [comment]
Here I think the distinction is mostly what is meant by "builder". Generally from the SLSA standpoint the "builder" is the CI system because that is what you trust to execute isolated "builds" so that's what I was referring to rather than a "builder" meaning a "build tool" like npm, docker, etc.

**Comment by ljharb** [comment]
I would be very confused if SLSA considered "the build" anything but the part of the pipeline that does transformation of source input, to released output - which is not what provenance provides any safety or guarantees about.

**Comment by ianlewis** [comment]
That's basically correct.
